[Inductor] Fix debug_str method of FusedSchedulerNode

kadeng opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #115877
* #115814
* #115813
* #115655
* #115654
* #112861
* #114687
* #115270
* #115174
* #115072
* #114606
* #114319
* #114075
* #113932
* #113891
* #113959
* #113890
* #113570
* #113569
* #113558
* #113399
* #113366
* __->__ #113365

Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

kadeng committed:
[Inductor] Fix debug_str method of FusedSchedulerNode

Calling debug_str on FusedSchedulerNode, which may happen in certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

Skylion007 commented on torch/_inductor/scheduler.py:
```diff
@@ -945,6 +945,26 @@ def allocate(self):
+                self.debug_str_extra(),
+            ]
+        except Exception:
+            log.warning("Ignoring error in debug_str()", exc_info=True)
```
Should this be a log.warning, or a logging.exception? Also can the exception type be any narrower here that BaseException?

kadeng replied:
I didn't want to change behavior here and just implement a minimal fix for an error. The entire method is almost an exact copy of the superclass method BaseSchedulerNode.debug_str, including this part. So if we change that, it should likely be changed in both. Apart from that, I agree with you, but for this diff I would prefer to keep the change minimal.

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng committed:
Update on "[Inductor] Fix debug_str method of FusedSchedulerNode"


Calling debug_str on FusedSchedulerNode, which may happen if certain
debug configuration options are enabled, results in an Exception
since self.node does not exist on FusedSchedulerNode.

This is a small fix to address that.

Test Plan:
CI

[ghstack-poisoned]

kadeng commented:
Moved to a (draft) feature branch, see https://github.com/pytorch/pytorch/pull/115919
